confidence/confidence.py

```diff
@@ -177,18 +152,46 @@ def _evaluate(
             flag_metadata={"flag_key": flag_key},
         )
 
-    def _resolve(self, flag_name: FlagName, context: Dict[str, str]) -> ResolveResult:
+    def track(self, event_name: str, context: Dict[str, str]) -> None:
```

Comment:
```suggestion
    def track(self, event_name: str, data: Dict[str, str]) -> None:
```
Also, data should be able to take more types..